api/envoy/config/filter/http/ext_authz/v2/ext_authz.proto

```diff
@@ -38,6 +38,17 @@ message ExtAuthz {
   // Note that errors can be *always* tracked in the :ref:`stats
   // <config_http_filters_ext_authz_stats>`.
   bool failure_mode_allow = 2;
+
+  // Enables filter to buffer and add the request body to the authorization request.
+  BufferSettings with_request_body = 4;
+}
+
+// Configuration for buffering the request data.
+message BufferSettings {
+  // Sets the maximum size of a message body that the filter will hold in memory. Envoy will return
+  // *HTTP 413* and will *not* initiate the authorization process when buffer reaches the number
```

Comment:
It might be orthogonal though, will there be some use cases that only require first certain number of bytes, so instead of 413, sending first x bytes to authz server?

Reply:
👍

Reply:
Still working on this one. Missing unit tests as well as some manual verification.